### PR TITLE
Update: Use single block editor in the widgets screen

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
All the other screens, we have used a single block editor. The widget screen used multiple block editors. The fact multiple block editors were used always presented additional complexity, e.g., we needed a system external to the block editor that tracks which widget area is active, and then we needed to use slots with bubblesvirtually to show UI pieces like the inserter bock inspector for the currently selected widget area.
On each feature we want to include, we would need to have some kind of custom implementation, e.g.: adding a block navigator for all the widget areas would evolve some complex slot fills to have it working.
I was trying to add the block navigator and tried to move them to a single block editor instead. The move proved to be very simple; we already have all the pieces to save the inner blocks of a block in a custom location, and it worked well. I ended up spending most of the time of this PR, fixing styling issues and handling things like events to unselect blocks, etc.
But things seem to be working well, and this PR ends up adding a new feature to the widget screen while removing 100+ lines of code.

The change also as some a11y advantages e.g., now we can use block editor select mode (ESC keypress) to navigate around widget areas, and tabbing is similar to other screens while previously tabbing was inconsistent and different from the other screens.

I think applying this change makes the widget screen even simpler and will make the future maintenance of the screen much easier is it now works like the other screens.

## How has this been tested?
I verified the widget screen still worked well, allowed saving, etc.
I verified I could use the block navigator on the widget screen.

### CODE REVIEW

File: packages/edit-widgets/src/components/widget-areas-block-editor-provider/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		setBlocks(
+			areas.map( ( { id, name }, index ) => {
+				return createBlock( 'core/widget-area', {
+					id,
+					name,
+					isOpen: index === 0,
+				} );
+			} )
+		);
+	}, [ areas ] );
```
Comment (ZebulanStanphill) [inline packages/edit-widgets/src/components/widget-areas-block-editor-provider/index.js:58]: Should `blocks` also be listed in the dependency array?

Comment (jorgefilipecosta) [reply packages/edit-widgets/src/components/widget-areas-block-editor-provider/index.js:58]: Nice catch @ZebulanStanphill, I think we should list it 👍

**Comment by jorgefilipecosta** [comment]
This PR is a dependency for some work we are doing on the widgets screen. I'm going to merge the PR so the work can advance more easily. In case there is any comment regarding a change done here please leave a comment and I will gladly address it as a follow-up PR.

File: packages/block-library/src/widget-area/edit/inner-blocks.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { InnerBlocks } from '@wordpress/block-editor';
+
+export default function WidgetAreaInnerBlocks() {
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'root',
+		'widgetArea'
+	);
+	return (
+		<InnerBlocks
+			__experimentalBlocks={ blocks }
```
Comment (noahtallen) [inline packages/block-library/src/widget-area/edit/inner-blocks.js:14]: This should be changed to `value` now that #21368 has been merged

**Review by noahtallen — COMMENTED** [review]
Hm, nothing stood out to me as problematic after seeing #22447.

File: packages/block-editor/src/components/inserter/index.js [hunk 3]
```diff
@@ -161,11 +161,14 @@ export default compose( [
...
 			hasInserterItems,
 			__experimentalGetAllowedBlocks,
+			getBlockSelectionEnd,
 		} = select( 'core/block-editor' );
 		const { getBlockVariations } = select( 'core/blocks' );
 
 		rootClientId =
-			rootClientId || getBlockRootClientId( clientId ) || undefined;
+			rootClientId ||
+			getBlockRootClientId( clientId || getBlockSelectionEnd() ) ||
```
Comment (talldan) [inline packages/block-editor/src/components/inserter/index.js:170]: #23263 proposes to revert this particular changes, as it caused the default root appender to insert blocks wherever selection is.

Comment (youknowriad) [reply packages/block-editor/src/components/inserter/index.js:170]: Seems logical @talldan the widget screen should  inject the rootClientId as a prop of the inserter instead.

**Pull request merged**